packages/agoric-cli/lib/deploy.js

```diff
@@ -223,12 +223,33 @@ export { bootPlugin } from ${JSON.stringify(absPath)};
             };
           }
 
-          // use a dynamic import to load the deploy script, it is unconfined
-          // eslint-disable-next-line import/no-dynamic-require
-          const mainNS = require(moduleFile);
-          // TODO Node.js ESM support if package.json of template says "type":
-          // "module":
-          //   const mainNS = await import(pathResolve(moduleFile));
+          // Use a dynamic import to load the deploy script.
+          // It is unconfined.
+
+          // Use Node.js ESM support if package.json of template says "type":
+          // "module".
+          const packageFile = pathResolve('package.json');
```

Comment:
Oh, another thing: the `package.json` should be searched starting from the `moduleFile`, since it is possible that the `moduleFile` is in a different subpackage than the top-level `package.json`.

Reply:
I resolved this by bringing in a utility in compartment mapper that searches for the package.json above any file.